pdns/opensslsigners.cc

```diff
@@ -910,6 +913,216 @@ void OpenSSLECDSADNSCryptoKeyEngine::fromPublicKeyString(const std::string& inpu
 }
 #endif
 
+#ifdef HAVE_LIBCRYPTO_EDDSA
+class OpenSSLEDDSADNSCryptoKeyEngine : public DNSCryptoKeyEngine
+{
+public:
+  explicit OpenSSLEDDSADNSCryptoKeyEngine(unsigned int algo) : DNSCryptoKeyEngine(algo)
+  {
+
+    int ret = RAND_status();
+    if (ret != 1) {
+      throw runtime_error(getName()+" insufficient entropy");
+    }
+
+    d_edkey = EVP_PKEY_new();
+    if (d_edkey == NULL) {
+      throw runtime_error(getName()+" allocation of key structure failed");
+    }
+
+    if(d_algorithm == 15) {
+      d_len = 32;
+      d_id = NID_ED25519;
+    } else if (d_algorithm == 16) {
+      d_len = 57;
+      d_id = NID_ED448;
+    } else {
+      throw runtime_error(getName()+" unknown algorithm "+std::to_string(d_algorithm));
+    }
+
+    d_edctx = EVP_PKEY_CTX_new_id(d_id, NULL);
```

Comment:
From what I see, it doesn't look like we need a `EVP_PKEY_CTX` object except for generating a new key so perhaps we could just deal with it in `::create()` and be done with it afterwards?